### PR TITLE
Add Notes and Questions for 19509

### PR DESCRIPTION
**Pull request by troygiorshev**
Two temp notes.

### CODE REVIEW

File: _posts/2020-08-19-#19509.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+## Notes
+
+* [PR 19509](https://github.com/bitcoin/bitcoin/pull/19509) is a feature PR, adding per-peer message logging to Bitcoin Core.
+
+* Nodes on the Bitcoin network communicate using a variety of [messages](https://btcinformation.org/en/developer-reference#p2p-network).  Now these messages can be logged for easy review, in order and on a per-peer basis.
+
+* The purpose of this feature is twofold: it should assist people in debugging their node, and it should help people who are unfamiliar with the Bitcoin network to get a feel of how everything works.
+
+* This feature is locked behind a debug-only `-logmessages` option.  When enabled, it writes every message the node recieves to a file, in a folder structure described in the PR.  It uses our `CAutoFile` [here](https://github.com/bitcoin/bitcoin/blob/ac953a3773aa5e940976a79276894f5e892d1c55/src/net.cpp#L2892).
```
Comment (jonatack) [inline _posts/2020-08-19-#19509.md]: s/recieves/receives/

File: _posts/2020-08-19-#19509.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+* The log parser lives in a new folder, at `contrib/message-logging/message-logging-parser.py`.  As discussed [here](https://github.com/bitcoin/bitcoin/pull/19509#issuecomment-664374225) it takes a list of files to parse, and optionally an output argument.
+
+* `LogMessage` takes the newly introduced `Span`, as discussed in a [previous PR review club](https://bitcoincore.reviews/18468).
```
Comment (jonatack) [inline _posts/2020-08-19-#19509.md]: can use a relative url for the link

File: _posts/2020-08-19-#19509.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+* [This commit](https://github.com/bitcoin/bitcoin/pull/19509/commits/ac953a3773aa5e940976a79276894f5e892d1c55) accounts for the increase in file descriptor usage.  This is due to be improved in [Issue 18911](https://github.com/bitcoin/bitcoin/issues/18911).
```
Comment (jonatack) [inline _posts/2020-08-19-#19509.md]: would suggest:

- s/this commit/commit ac953a3/ both here and line 37

- and maybe adding the titles of the commit and the issue so people can see their names without having to click through on first read

File: _posts/2020-08-19-#19509.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+## Questions
+
+1. Did you review the PR? [Concept ACK, approach ACK, tested ACK, or
+   NACK](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#peer-review)?
+   (You're always encouraged to put your PR review on GitHub, even after the PR has been merged.)
+
+2. How did you test this PR?  Was setting up and running everything straightforward enough?  Reminder that the test runner has a `--nocleanup` option.
```
Comment (jonatack) [inline _posts/2020-08-19-#19509.md]: suggest s/Reminder/As a reminder,|Remember/

File: _posts/2020-08-19-#19509.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+3. Did you verify that file descriptor accounting was being done correctly in [this commit](https://github.com/bitcoin/bitcoin/pull/19509/commits/ac953a3773aa5e940976a79276894f5e892d1c55)?  What is the soft file desciptor limit on your system?  What about the hard limit?  (TEMP NOTE: I'm happy to lead them into `ulimit` here in the notes, not sure where to draw the line.  Don't want people just blindly running commands)
+
+4. [The first commit](https://github.com/bitcoin/bitcoin/pull/19509/commits/b3a0e528d1128ee456258f4b265b52b7e8fbc6a6) is a cleanup.  Is this justified or is this just noise?  When should cleanups like this be done?
+
+5. [This commit](https://github.com/bitcoin/bitcoin/pull/19509/commits/4fc2e3906377991a05f78a938c0aa2f04056b81c) adds a test.  What is being tested?  What isn't being tested?  Should this be expanded?  Did you read the out-of-tree build discussion [here](https://github.com/bitcoin/bitcoin/pull/19509#discussion_r465770624)?
```
Comment (jonatack) [inline _posts/2020-08-19-#19509.md]: As above, I find "This commit" here and line 39 to be rather generic (these are different commits) and unhelpful (no commit hash, no name/title).

File: _posts/2020-08-19-#19509.md [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+6. Both [jnewbery](https://github.com/bitcoin/bitcoin/pull/19509#pullrequestreview-447894982) and [practicalswift](https://github.com/bitcoin/bitcoin/pull/19509#pullrequestreview-448269477) proposed extensions to this message logging. What are they? Do you agree that these are a good idea? Are there any extensions that you would like to see?
```
Comment (jonatack) [inline _posts/2020-08-19-#19509.md]: suggest jnewbery [comment](url) (same for practicalswift) instead to make clear that the links are to comments and not to the github profiles.

File: _posts/2020-08-19-#19509.md [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+host: troygiorshev
+status: upcoming
+commit: 4fc2e3906
+---
+
+## Notes
+
+* [PR 19509](https://github.com/bitcoin/bitcoin/pull/19509) is a feature PR, adding per-peer message logging to Bitcoin Core.
+
+* Nodes on the Bitcoin network communicate using a variety of [messages](https://btcinformation.org/en/developer-reference#p2p-network).  Now these messages can be logged for easy review, in order and on a per-peer basis.
```
Comment (jonatack) [inline _posts/2020-08-19-#19509.md]: maybe s/Now/This PR proposes to permit/ and s/order/chronological order/ (if that's the case)

File: _posts/2020-08-19-#19509.md (hunk 1 above)
Comment (jnewbery) [reply _posts/2020-08-19-#19509.md]: fixed

File: _posts/2020-08-19-#19509.md (hunk 2 above)
Comment (jnewbery) [reply _posts/2020-08-19-#19509.md]: done

File: _posts/2020-08-19-#19509.md (hunk 3 above)
Comment (jnewbery) [reply _posts/2020-08-19-#19509.md]: I've changed all commit links to explicitly name the commits they're linking to.

File: _posts/2020-08-19-#19509.md (hunk 4 above)
Comment (jnewbery) [reply _posts/2020-08-19-#19509.md]: done

File: _posts/2020-08-19-#19509.md (hunk 5 above)
Comment (jnewbery) [reply _posts/2020-08-19-#19509.md]: Fixed

File: _posts/2020-08-19-#19509.md (hunk 6 above)
Comment (jnewbery) [reply _posts/2020-08-19-#19509.md]: done

File: _posts/2020-08-19-#19509.md (hunk 7 above)
Comment (jnewbery) [reply _posts/2020-08-19-#19509.md]: reworded

**Comment by jnewbery** [comment]
Thanks for the review @jonatack . I've addressed all your review comments.

@troygiorshev: I think both of the notes that you marked TEMP are good and should stay.

**Comment by jonatack** [comment]
Nice timing.

**Comment by jonatack** [comment]
Tested ACK, LGTM

**Comment by troygiorshev** [comment]
Posthumous ACK to the changes :D

**Pull request merged**